recipe/meta.yaml

```diff
@@ -43,12 +43,7 @@ requirements:
     - libgomp      # [linux or win]
     - ninja
   host:
-    ############################################################################################
-    # Pin to CCCL 2.7 to avoid build issues in older or newer CCCLs.                           #
-    #                                                                                          #
-    # xref: https://github.com/conda-forge/xgboost-feedstock/pull/219#issuecomment-2738813028  #
-    ############################################################################################
-    - cccl =2.7             # [cuda_compiler_version != "None"]
+    - cccl                  # [cuda_compiler_version != "None"]
```

Comment:
CCCL 3.0 is now supported in XGBoost 3.0.3 thanks to PR: https://github.com/dmlc/xgboost/pull/11583

Also the latest CCCL may have fixed the past issues we saw when building with CCCL 2.8

So let's try unpinning CCCL to pick up